src/lib/validation.ts

```diff
@@ -12,6 +12,7 @@ export interface CheckIntOptions {
 
 const ENV_VAR_KEY_REGEX = /^[a-zA-Z_][a-zA-Z0-9_]*$/;
 const LABEL_NAME_REGEX = /^[a-zA-Z][a-zA-Z0-9\.\-]*$/;
+const NUMERALS_REGEX = /^-?[0-9]*\.?0*$/; // Allows trailing 0 decimals
```

Comment:
This feels very adhoc to me, and something we shouldn't really be implementing. I think that https://lodash.com/docs/4.17.15#toNumber should cover what we want, could you check that please?

Reply:
The problem is all these conversion functions are pretty generous with what they accept. We however want to allow only a very specific input so the regex was the easiest way to test that then use the conversion functions to only verify and cast to Number type. 

If I remove that regex then I have to write a regex to catch cases like '0x11', '0b11', '0o11' since [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) will accept them and convert to int. Lodash's implementation behaves the same.

Reply:
We agreed to stick with regex as that was how others (mdn specifically) implemented it as well.